docs/content/guides/dagster/branch_depl_with_snowflake.mdx

```diff
@@ -0,0 +1,325 @@
+4. Delete the corresponding database clone upon closing the feature branch.
+
+### Step 1: Create our job
+
+In production, we want to write three tables to Snowflake: items, comments, and stories. We can define these tables as [assets](https://docs.dagster.io/concepts/assets/software-defined-assets) as follows:
+
+```python file=/guides/dagster/local_to_prod_tutorial/assets.py startafter=start_assets endbefore=end_assets
+# assets.py
+
+
+@asset(
+    config_schema={"N": int},
+    required_resource_keys={"hn_client"},
+)
+def items(context) -> pd.DataFrame:
+    """Items from the Hacker News API: each is a story or a comment on a story."""
+    hn_client = context.resources.hn_client
+
+    rows = []
+    # Hacker News API is 1-indexed, so adjust range by 1
+    for item_id in range(context.op_config["N"] + 1, hn_client.fetch_max_item_id() + 1):
+        rows.append(hn_client.fetch_item_by_id(item_id))
+
+    result = pd.DataFrame(rows, columns=hn_client.item_field_names).drop_duplicates(
+        subset=["id"]
+    )
+    result.rename(columns={"by": "user_id"}, inplace=True)
+    return result
+
+
+@asset
+def comments(items: pd.DataFrame) -> pd.DataFrame:
+    """Comments from the Hacker News API."""
+    return items[items["type"] == "comment"]
+
+
+@asset
+def stories(items: pd.DataFrame) -> pd.DataFrame:
+    """Stories from the Hacker News API."""
+    return items[items["type"] == "story"]
+```
+
+As you can see, our assets use a [resource](https://docs.dagster.io/\_apidocs/libraries/dagster-snowflake#dagster_snowflake.snowflake_resource) named `hn_client` (source code available in the [Transitioning Data Pipelines from Development to Production](/guides/dagster/local-to-production-tutorial#part-one-local-development) guide). Using a resource allows us to swap out implementations per environment without modifying our computational code.
+
+### Step 2: Configure our job for each environment
+
+At runtime, we’d like to determine which environment our code is running in: branch deployment, or production. This is valuable information that will dictate how our code should execute—with which credentials and with which database. We’ll use a different set of credentials in each environment to ensure we can’t accidentally write to production within our branch deployment, and we’ll write to our database clone in our branch deployment instead of `PRODUCTION`.
+
+Dagster automatically sets certain environment variables containing deployment metadata, allowing us to read these environment variables to discern between deployments. For a full list of preset environment variables, visit this link. We can access the `DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT` environment variable to determine the currently executing environment.
+
+Because we want to configure our assets to write to Snowflake using a different set of credentials and database in each environment, we’ll configure a separate IO manager for each environment:
```

Comment:
```suggestion
Because we want to configure our assets to write to Snowflake using a different set of credentials and database in each environment, we’ll configure a separate I/O manager for each environment:
```

[2]
Making this `I/O` as per a recent style guide decision - I know I pointed this out on another docs PR of yours recently where it was `IO`. This is the last time it changes!